lib/text_parser.dart

```diff
@@ -146,81 +158,35 @@ class TextParser extends StatelessWidget {
     return LinkTextSpans(
       text: finalText,
       themeData: Theme.of(context),
-      onLinkTap: onLinkTap,
+      beforeLaunch: beforeLaunch ??
+          // ignore: deprecated_member_use_from_same_package
+          (onLinkTap != null
+              ? (uri) {
+                  // ignore: deprecated_member_use_from_same_package
+                  onLinkTap?.call(uri.toString());
+                  return false;
+                }
+              : null),
       textStyle: parseContext.textStyle,
       linkStyle: parseContext.textStyle.merge(parseContext.linkStyle),
+      hoverStyle: parseContext.textStyle.merge(parseContext.linkHoverStyle),
     );
   }
 
   InlineSpan _optimizeTextspan(InlineSpan textSpan) {
-    if (!(textSpan is TextSpan)) {
+    if (textSpan is! TextSpan) {
       return textSpan;
     }
     // if we have only one child and inherit all styles...return it
     if ((textSpan.text?.isEmpty ?? true) &&
         textSpan.children?.length == 1 &&
-        (textSpan.style == null || textSpan.style == TextStyle()) &&
+        (textSpan.style == null || textSpan.style == const TextStyle()) &&
         (textSpan.recognizer == null ||
             (textSpan.children!.first is TextSpan &&
                 (textSpan.children!.first as TextSpan).recognizer ==
                     textSpan.recognizer))) {
       return _optimizeTextspan(textSpan.children!.first);
     }
-    // if our child node is just blank, then append its child nodes one up
```

Comment:
why did you just remove this block?

Reply:
Because those operations are no longer permitted in Flutter 3.0.0. I found no way to workaround it yet.

Reply:
Which ones specifically?

Reply:
all the direct manipulation of `.children`?

Reply:
if so, since it is just child clearing and then add all, perhaps it could create a new TextSpan instead with the new children, and copying the style